types/v56/types.go

```diff
@@ -445,49 +445,96 @@ type AdminVdc struct {
 // Namespace: http://www.vmware.com/vcloud/v1.5
 // Description: Parameters for creating an organization VDC
 // Since: 5.1
 // https://code.vmware.com/apis/220/vcloud#/doc/doc/types/CreateVdcParamsType.html
 type VdcConfiguration struct {
-	XMLName                  xml.Name             `xml:"CreateVdcParams"`
-	Xmlns                    string               `xml:"xmlns,attr"`
-	Name                     string               `xml:"name,attr"`
-	Description              string               `xml:"Description,omitempty"`
-	AllocationModel          string               `xml:"AllocationModel"` // Flex supported from 32.0
-	ComputeCapacity          []*ComputeCapacity   `xml:"ComputeCapacity"`
-	NicQuota                 int                  `xml:"NicQuota,omitempty"`
-	NetworkQuota             int                  `xml:"NetworkQuota,omitempty"`
-	VmQuota                  int                  `xml:"VmQuota,omitempty"`
-	IsEnabled                bool                 `xml:"IsEnabled,omitempty"`
-	VdcStorageProfile        []*VdcStorageProfile `xml:"VdcStorageProfile"`
-	ResourceGuaranteedMemory *float64             `xml:"ResourceGuaranteedMemory,omitempty"`
-	ResourceGuaranteedCpu    *float64             `xml:"ResourceGuaranteedCpu,omitempty"`
-	VCpuInMhz                int64                `xml:"VCpuInMhz,omitempty"`
-	IsThinProvision          bool                 `xml:"IsThinProvision,omitempty"`
-	NetworkPoolReference     *Reference           `xml:"NetworkPoolReference,omitempty"`
-	ProviderVdcReference     *Reference           `xml:"ProviderVdcReference"`
-	UsesFastProvisioning     bool                 `xml:"UsesFastProvisioning,omitempty"`
-	OverCommitAllowed        bool                 `xml:"OverCommitAllowed,omitempty"`
-	VmDiscoveryEnabled       bool                 `xml:"VmDiscoveryEnabled,omitempty"`
-	IsElastic                *bool                `xml:"IsElastic,omitempty"`             // Supported from 32.0 for the Flex model
-	IncludeMemoryOverhead    *bool                `xml:"IncludeMemoryOverhead,omitempty"` // Supported from 32.0 for the Flex model
+	XMLName                  xml.Name                   `xml:"CreateVdcParams"`
+	Xmlns                    string                     `xml:"xmlns,attr"`
+	Name                     string                     `xml:"name,attr"`
+	Description              string                     `xml:"Description,omitempty"`
+	AllocationModel          string                     `xml:"AllocationModel"` // Flex supported from 32.0
+	ComputeCapacity          []*ComputeCapacity         `xml:"ComputeCapacity"`
+	NicQuota                 int                        `xml:"NicQuota,omitempty"`
+	NetworkQuota             int                        `xml:"NetworkQuota,omitempty"`
+	VmQuota                  int                        `xml:"VmQuota,omitempty"`
+	IsEnabled                bool                       `xml:"IsEnabled,omitempty"`
+	VdcStorageProfile        []*CreateVdcStorageProfile `xml:"VdcStorageProfile"`
```

Comment:
This is getting confusing. Please add a comment why "CreateVdcStorageProfile" in "VdcStorageProfile".

Reply:
it was hard to grasp for me, until I figured out that API uses 3 types or VDC storage profiles which are almost incidental. Some of them need different XML root name. Every type has links to API description type. In general CreateType is used when creating Org VDC - which has only a few fields, VdcStorageProfile is used for GET and AdminVdcStorageProfile for update.

Reply:
Please add the explanation as comment in code.

Reply:
Where too? all this type are referenced in different locations and every type already has explanation for what it used and reference to API.

Reply:
Could we possible call it `AttachVdcStorageProfile` or something similar. The `CreateVdcStorageProfile` sounds as if it is going to create a storage profile but in fact it is not (storage profiles are defined in pVdcs)

Reply:
Well any more better namings? I understand your reasoning. I tried to match to createOrgVdc logic. If other agrees and there isn't better ides - I will change. @datacharmer @lvirbalas

Reply:
The current naming is not acceptable because is a double breaking change.
We are renaming an existing structure (`VdcStorageProfile`) to `CreateVdcStorageProfile`, and what is worse, we are giving the old name `VdcStorageProfile` to a new structure.

Reply:
Well previously existing type usage wasn't correct - it was bug. I made changes to have least impact as possible. Func interfaces haven't changed. Only one thing changes is that `OrgVdc` internal Type `VdcStorageProfile` changed to `CreateVdcStorageProfile` .

Reply:
The right procedure in this case would be to deprecate the existing structure and creating a new one. As it is we are introducing two levels of confusion with this change.

Reply:
Well result will be I have to introduce new functions for createOrgVdc, createOrgVdcType then +4 VdcStorageProfile types and new functions for getStorageProfilHref and everything will be with "silly namings" and deprecate 2 functions. Overall this minimum I recalled in my head. @lvirbalas  what to do?

Reply:
After discussion we renamed to `VdcStorageConfiguration` to be inline with `VdcConfiguration` - structs that are used to **create** the VDC.

@dataclouder do you have an alternative proposal without deprecations? One alternative I can think of is actually leaving old name as is, but adding a suffix or prefix "Read" to the new structs:
`VdcStorageProfileRead` and `AdminVdcStorageProfileRead`.

Either way, we need to be consistent with these kind of prefixes going forward.

Reply:
if we do that, we break get function definitions. So more breakage

Reply:
>leaving old name as is, but adding a suffix or prefix "Read" to the new structs:
VdcStorageProfileRead and AdminVdcStorageProfileRead.

That I could live with. And it would make it possible to deprecate the old one.

Reply:
Let's go with `VdcStorageProfileConfiguration`, just last ask to add breaking changes section to changelog.